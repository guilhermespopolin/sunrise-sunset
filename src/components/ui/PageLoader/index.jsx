import React from 'react'
import PropTypes from 'prop-types'

import Overlay from 'components/ui/Overlay'
import Loading from 'components/ui/Loading'

function PageLoader({ error, pastDelay, timeout }) {
  if (error) {
    return <h1>Something went wrong!</h1>
  } else if (timeout) {
    return <h1>Is taking a long time...refresh page</h1>
  } else if (pastDelay) {
    // should render loading component.
    // avoid flash of loading component
    return (
      <Overlay>
        <Loading />
      </Overlay>
    )
  }

  return null
}

PageLoader.propTypes = {
  error: PropTypes.instanceOf(Error),
  pastDelay: PropTypes.bool,
  timeout: PropTypes.bool,
}

PageLoader.defaultProps = {
  error: null,
  pastDelay: false,
  timeout: false,
}

export default PageLoader
