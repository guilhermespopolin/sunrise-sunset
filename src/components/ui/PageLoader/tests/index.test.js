import React from 'react'
import { render } from '__tests__/utils/rtlWrapper'

import PageLoader from '..'

describe('<PageLoader />', () => {
  describe('When some error occur during page loading', () => {
    it('should render properly', () => {
      const { container } = render(<PageLoader error={new Error('Something went wrong!')} />)

      expect(container.firstChild).toMatchSnapshot()
    })
  })

  describe('When timeout occur during page loading', () => {
    it('should render properly', () => {
      const { container } = render(<PageLoader timeout />)

      expect(container.firstChild).toMatchSnapshot()
    })
  })

  describe('When page takes considerable amount of time to be loaded', () => {
    it('should render properly', () => {
      const { container } = render(<PageLoader pastDelay />)

      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
