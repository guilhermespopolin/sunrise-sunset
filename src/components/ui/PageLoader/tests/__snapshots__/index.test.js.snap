// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLoader /> When page takes considerable amount of time to be loaded should render properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: rgba(0,0,0,0.8);
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c1 {
  display: inline-block;
  width: 15px;
  height: 15px;
  border-radius: 15px;
  background-color: #ff9900;
}

.c1[data-id="1"] {
  -webkit-animation: ioOCvU 0.6s 0.1s linear infinite;
  animation: ioOCvU 0.6s 0.1s linear infinite;
}

.c1[data-id="2"] {
  -webkit-animation: ioOCvU 0.6s 0.2s linear infinite;
  animation: ioOCvU 0.6s 0.2s linear infinite;
}

.c1[data-id="3"] {
  -webkit-animation: ioOCvU 0.6s 0.3s linear infinite;
  animation: ioOCvU 0.6s 0.3s linear infinite;
}

.c1:not(:last-child) {
  margin-right: 0.5rem;
}

<div
  class="c0"
  display="flex"
>
  <div
    class=""
    data-testid="loading"
  >
    <div
      class="c1"
      data-id="1"
    />
    <div
      class="c1"
      data-id="2"
    />
    <div
      class="c1"
      data-id="3"
    />
  </div>
</div>
`;

exports[`<PageLoader /> When some error occur during page loading should render properly 1`] = `
<h1>
  Something went wrong!
</h1>
`;

exports[`<PageLoader /> When timeout occur during page loading should render properly 1`] = `
<h1>
  Is taking a long time...refresh page
</h1>
`;
